### PR TITLE
DEV#233, check CRUD

### PR DESCRIPTION
**Pull request by dhMuse**

### CODE REVIEW

File: app/helpers/reports_helper.rb [hunk 1]
```diff
@@ -90,4 +90,17 @@ def report_index_dropdown(report_type)
         end.join.html_safe
     end
   end
+
+  def self.included_entities(specialization, divisions, entity)
+    if divisions.length == 1
```
Comment (bgracie) [inline app/helpers/reports_helper.rb]: `.one?`

File: app/helpers/reports_helper.rb [hunk 2]
```diff
@@ -90,4 +90,17 @@ def report_index_dropdown(report_type)
...
+      specialization.send(entity).in_divisions(divisions)
+    else
+      specialization.send(entity)
+    end.reject do |entity|
+      !entity.show_waittimes? ||
+        if entity.respond_to?(:unavailable_for_a_while?)
+          entity.unavailable_for_a_while?
```
Comment (bgracie) [inline app/helpers/reports_helper.rb]: the implicit `nil` returned if the condition isn't satisfied is confusing my brain

Comment (bgracie) [inline app/helpers/reports_helper.rb]: try `select do |entity| entity.show_waittimes && (!entity.respond_to?(:unavailable_for_a_while?) || !entity.unavailable_for_a_while?) end`

File: app/models/clinic.rb [hunk 3]
```diff
@@ -98,6 +98,12 @@ class Clinic < ActiveRecord::Base
   has_many :clinic_healthcare_providers, dependent: :destroy
   has_many :healthcare_providers, through: :clinic_healthcare_providers
 
+  has_many :favorites, as: :favoritable, dependent: :destroy
+  has_many :favorite_users,
```
Comment (bgracie) [inline app/models/clinic.rb]: I would say `favoriting_users`, here and in the other models.  clinics, sc_items, and specialists can't favorite anything.

File: app/helpers/reports_helper.rb (hunk 2 above)
Comment (dhMuse) [inline app/helpers/reports_helper.rb]: Yeah, I don't know why I didn't just do it that way right off ...

File: app/models/clinic.rb (hunk 3 above)
Comment (dhMuse) [inline app/models/clinic.rb]: Agreed. I was respecting existing conventions, but there's really no reason to.

**Pull request merged**